src/diffusers/loaders/lora_pipeline.py

```diff
@@ -1982,9 +1982,19 @@ def _maybe_expand_transformer_param_shape_or_error_(
                 out_features = state_dict[lora_B_weight_name].shape[0]
 
                 # This means there's no need for an expansion in the params, so we simply skip.
-                if tuple(module_weight.shape) == (out_features, in_features):
+                module_weight_shape = module_weight.shape
+                expansion_shape = (out_features, in_features)
+                quantization_config = getattr(transformer, "quantization_config", None)
+                if quantization_config and quantization_config.quant_method == "bitsandbytes":
+                    if quantization_config.load_in_4bit:
+                        expansion_shape = torch.Size(expansion_shape).numel()
+                        expansion_shape = ((expansion_shape + 1) // 2, 1)
```

Comment:
Only 4bit bnb models flatten.

Reply:
How about adding a comment along the lines of: "Handle 4bit bnb weights, which are flattened and compress 2 params into 1".

I'm not quite sure why we need (shape+1) // 2, maybe this could be added to the comment too.

Reply:
Yeah, this comes from `bitsandbytes`. Cc: @matthewdouglas

Reply:
This is for rounding, i.e. if `expansion_shape` is odd it will have an additional 8bit tensor with just one value packed into it.